### PR TITLE
js files

### PR DESCRIPTION
**Pull request by lawcia**

### CODE REVIEW

**Comment by ssahabg** [comment]
@lawcia - Looks good.
If you add few things like editor config file (https://editorconfig.org/), ESlint config file, that would be good.

File: spec/Task4Spec.js [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+      expect(function(){ sortObjArray(0, []); }).toThrow(new Error("Parameter key must be of type string, and arr must be an object"));
+      expect(function(){ sortObjArray("hello", 0); }).toThrow(new Error("Parameter key must be of type string, and arr must be an object"));
+    });
+  
+    it("the key parameter should be a key in object", () => {
+      expect(function(){ sortObjArray("name", [{"age": 1}, {"age": 2}]); }).toThrow(new Error("Objects in array must have property key"));
+    });
+  
+    it("should sort by title", () => {
+      let arr = [
```
Comment (ssahabg) [inline spec/Task4Spec.js:13]: Hi @lawcia - Try to keep test data in separate file and reusing it. This would help a lot.

File: spec/Task5Spec.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      const animal = {
+        name: "Andy",
+        age: 10
+      }
+  
+      const mammal = {
+        furColour: "Brown"
+      }
+  
+      mammal.__proto__ = animal;
```
Comment (ssahabg) [inline spec/Task5Spec.js:13]: @lawcia - Try not to override `__proto__` instead refer to https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Objects/Object_prototypes

File: spec/Task5Spec.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      const mammal = {
+        furColour: "Brown"
+      }
+  
+      const dog = {
+        tail: true
+      }
+  
+      mammal.__proto__ = animal;
+      dog.__proto__ = mammal;
```
Comment (ssahabg) [inline spec/Task5Spec.js:56]: Same as above

**Comment by lawcia** [comment]
I have made the required changes

File: src/Task5.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+// Task 5
+// Write a JavaScript function to retrieve all the names of object's own and inherited properties
+function getObjectNames(obj) {
+    let properties = [];
+
+    for (let key in obj) {
+        properties.push(key);
```
Comment (pradeepchalla01) [inline src/Task5.js:7]: this will give all the properties in the object, can you differentiate between inherited and own properties

**Review by pradeepchalla01 — APPROVED** [review]

**Review by ssahabg — APPROVED** [review]

**Pull request merged**